packages/shared/src/components/modals/common.tsx

```diff
@@ -44,7 +49,7 @@ type GetComponentProps<T> = T extends
   ? P
   : never;
 
-type ModalsType = typeof modals;
+export type ModalsType = typeof modals;
```

Comment:
This shouldn't need an export, as the LazyModal types already have their own?

Reply:
We need it in the `LazyModal` to make typescript with `data` in the `useQuery` work. Currently the `data` when accessed returns unknown and make any attempt to use its property, not work.

Reply:
Hmm but what if you use the initial type we already export?

Reply:
Yes, that is what it is using initially. Had to make it accept a type parameter.